server/src/main/java/io/druid/segment/loading/StorageLocation.java

```diff
@@ -29,17 +30,33 @@
 */
 class StorageLocation
 {
+  private static final Logger log = new Logger(StorageLocation.class);
+
   private final File path;
   private final long maxSize;
+  private final long freeSpaceToKeep;
   private final Set<DataSegment> segments;
 
   private volatile long currSize = 0;
 
-  StorageLocation(File path, long maxSize)
+  StorageLocation(File path, long maxSize, Float freeSpacePercent)
```

Comment:
1) Please use double. float multiplied by long (`freeSpacePercent * totalSpaceInPartition`) is converted to float, that may have pretty bad precision, when `maxSize` is big.

2) Add `@Nullable`

Reply:
changed